lib/blurhash/ruby.rb

```diff
@@ -0,0 +1,170 @@
+module Blurhash
+  # :stopdoc:
+  module Ruby
+    class ThreeDArray
+      def initialize(y, x, z)
+        @y = y
+        @x = x
+        @z = z
+        @list = Array.new(y * x * z)
+      end
+
+      def set(y, x, z, val)
+        i = z + (x * @z) + (y * @z * @x)
```

Comment:
Hi. I'm just randomly subscribed to this repo and was interested by seeing such pull request.
Just wanted to say that IIRC this work do any profit in Ruby, you better just have real 3d array. Not sure if there are benchmarks here but I bet on this.

Reply:
I don't really want to rewrite this to use a "real 3d array".  You're welcome to do it and test the performance.  My guess is that they'd be pretty much the same, or this version is faster.  Why?

1. A real 3d array will have many more objects, causing more time in GC
2. A real 3d array will require more pointer chasing (we would have to dereference more objects)

In the presence of a JIT compiler, this version will no doubt be faster (math is much cheaper than reading memory).  Again, you're welcome to test!